cli/src/test/all-tests-functions.sh

```diff
@@ -43,22 +43,40 @@ function doDownloadSnapshot () {
     local url
     # Change OS type based on github workflow matrix.os name
     local osType
-    if [ "${MATRIX_OS}" == "windows-latest" ]; then
-      osType="windows-x64"
-    elif [ "${MATRIX_OS}" == "ubuntu-latest" ]; then
-      osType="linux-x64"
-    elif [ "${MATRIX_OS}" == "macos-latest" ]; then
-      osType="mac-arm64"
-    elif [ "${MATRIX_OS}" == "macos-13" ]; then
-      osType="mac-x64"
-    fi
-    url=$(grep "href=\"https://.*${osType}.tar.gz" "$pageHtmlLocal" | grep -o "https://.*${osType}.tar.gz" | cut -f1 -d"\"")
+    osType=$(doGetOsType)
+    url=$(grep "href=\"https://.*${osType}.tar.gz" "$pageHtmlLocal" | grep -o "https://[^\"]*${osType}.tar.gz" | head -1)
     echo "Trying to download IDEasy for OS: ${osType} from: ${url} to: ${IDEASY_COMPRESSED_FILE:?} ..."
     curl -o "${IDEASY_COMPRESSED_FILE:?}" "$url"
     rm "${pageHtmlLocal:?}"
   fi
 }
 
+function doGetOsType() {
+  local osType
+  case "$OSTYPE" in
+    msys*|cygwin*|win32*)
+      osType="windows-x64"
```

Comment:
This is actually wrong. I tested this on Windows@ARM64 and it gives me `windows-x64`.
However, the architecture does not seem to matter in any of our integration tests.
IMHO this is somehow coming from the replaced `MATRIX_OS`.
I would suggest that we simply remove the architecture since the method names are only about the operating system: `doGetOsType`, `doIsWindows`, `doIsMacOs`.
If we later need a separation by architecture, we can add another function for that.

Reply:
I created issue #1665 for this topic.
I therefore resolve this now so we can merge this PR and address that issue later.